src/common/iop_order.c

```diff
@@ -0,0 +1,2418 @@
+/*
+    This file is part of darktable,
+    copyright (c) 2018 edgardo hoszowski.
+
+    darktable is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    darktable is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "common/darktable.h"
+#include "common/colorspaces_inline_conversions.h"
+#include "common/iop_order.h"
+#include "develop/imageop.h"
+#include "develop/imageop_math.h"
+#include "develop/pixelpipe.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define DT_IOP_ORDER_VERSION 2
+
+static void _ioppr_insert_iop_after(GList **_iop_order_list, GList *history_list, const char *op_new, const char *op_previous, const int dont_move);
+static void _ioppr_insert_iop_before(GList **_iop_order_list, GList *history_list, const char *op_new, const char *op_next, const int dont_move);
+static void _ioppr_move_iop_after(GList **_iop_order_list, const char *op_current, const char *op_prev, const int dont_move);
+static void _ioppr_move_iop_before(GList **_iop_order_list, const char *op_current, const char *op_next, const int dont_move);
+
+
+/* migrates *_iop_order_list from old_version to the next version (version + 1)
+ * limitations:
+ * - to move an existing module that is always enabled a new version must be created, otherwise
+ *   modules can be added/moved in the current version
+ * - a module can't be more than once on the same version
+ */
+static int _ioppr_legacy_iop_order_step(GList **_iop_order_list, GList *history_list, const int old_version, const int dont_move)
+{
+  int new_version = -1;
+  
+  if(0)
+  {
+    _ioppr_insert_iop_before(_iop_order_list, history_list, "dummy1", "colorin", dont_move);
```

Comment:
what is dummy1 and dummy2? was this for debugging? if it is the case shouldn't it be removed?